libs/common/src/sql-db-entities/fee-record/fee-record.entity.test.ts

```diff
@@ -107,4 +107,84 @@ describe('FeeRecordEntity', () => {
       expect(feeRecord.lastUpdatedByTfmUserId).toBe(userId);
     });
   });
+
+  describe('markAsReconciled', () => {
+    const mockDate = new Date('2024-01-01');
+
+    beforeAll(() => {
+      jest.useFakeTimers();
+    });
+
+    beforeEach(() => {
+      jest.setSystemTime(mockDate);
+    });
+
+    afterAll(() => {
+      jest.useRealTimers();
+    });
+
+    it('sets the fee record dateReconciled to now and the reconciledByUserId to the supplied value', () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withDateReconciled(null).build();
+
+      // Act
+      feeRecord.markAsReconciled({ reconciledByUserId: 'abc123', requestSource: { platform: 'TFM', userId: 'def456' } });
+
+      // Assert
+      expect(feeRecord.dateReconciled).toEqual(mockDate);
+      expect(feeRecord.reconciledByUserId).toBe('abc123');
+    });
+
+    it("sets the fee record status to RECONCILED and updates the 'lastUpdatedBy...' fields", () => {
+      // Arrange
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport)
+        .withStatus(FEE_RECORD_STATUS.READY_TO_KEY)
+        .withLastUpdatedByIsSystemUser(true)
+        .withLastUpdatedByPortalUserId(null)
+        .withLastUpdatedByTfmUserId(null)
+        .build();
+
+      // Act
+      feeRecord.markAsReconciled({ reconciledByUserId: 'abc123', requestSource: { platform: 'TFM', userId: 'abc123' } });
+
+      // Assert
+      expect(feeRecord.status).toBe(FEE_RECORD_STATUS.RECONCILED);
+      expect(feeRecord.lastUpdatedByIsSystemUser).toBe(false);
+      expect(feeRecord.lastUpdatedByPortalUserId).toBeNull();
+      expect(feeRecord.lastUpdatedByTfmUserId).toBe('abc123');
```

Comment:
Can we test the full result (`feeRecord` object) rather than specific fields?

Reply:
You can't test a class instance the same way you would an object in terms of checking all the fields due to the class methods, hence only checking specific fields

Reply:
Can we please double check all the fields that returned by each class are testing every appropriate property?

For example in line 148, it calls `markAsReconciled`. This returns the following:

```ts
reconciledByUserId
updateLastUpdatedBy
dateReconciled
status
```

But in the test, we're only testing these:

```ts
status
lastUpdatedByIsSystemUser
lastUpdatedByPortalUserId
lastUpdatedByTfmUserId
```

Reply:
A previous test in the describe block tests that the other properties are updated